Project-Lily/Project-Lily/NetworkManager.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  NetworkManager.swift
+//  Project-Lily
+//
+//  Created by Kendall Poindexter on 11/16/18.
+//  Copyright © 2018 Kendall Poindexter. All rights reserved.
+//
+
+import Foundation
+
+class NetworkManager {
+    
+    private let urlString = "https://private-52aac-breeds1.apiary-mock.com/breeds"
+    
+    func createDogURLSession(completion: @escaping ([Dog]) -> Void) {
+        let session = URLSession.shared
+        guard let url = createDogURL(urlString: urlString) else { return }
+        let dataTask = session.dataTask(with: url) { (data, response, error) in
+            if let error = error {
+                print("Failure! \(error)")
+            } else if let httpResponse = response as? HTTPURLResponse,
+                httpResponse.statusCode == 200 {
+                if let data = data {
+                    let dogs: [Dog] = self.parse(data: data) ?? []
+                   
+                    
+                    DispatchQueue.main.async {
+                         completion(dogs)
+                        //self.dogListTableView.reloadData()
+                    }
+                }
+                return
+            } else {
+                print("Failure! \(response!)")
+            }
+            
+        }
+        
+        dataTask.resume()
+    }
+    
+   private func createDogURL(urlString: String) -> URL? {
+        guard let url = URL(string: urlString) else {return nil}
```

Comment:
The `createDogURL` function is effectively pointless here since it returns an optional. That is because the `URL(string:)` constructor also returns an optional. For example, what's the difference between this:

```swift
let url = createDogURL(urlString: urlString)
```

And this?:

```swift
let url = URL(string: urlString)
```

Take a look at line 17 in your code above and see where you can optimize it.